README.md

```diff
@@ -1,31 +1,198 @@
-
-For using the client/server python scripts they work in pairs. The UDP
-Client script works with the UDP Server script and the TCP Client script
-works with the TCP Server script. The client/server scripts require some
-manual configuration within the scripts. You will have to put the
-destination IP(s) at “server = [your.server.goes.here]”. The port that
-the scripts are running on by default is 9000 but it can be changed at
-the line with “PORT = 9000” or “SERVER_PORT = 9000”. This script should
-scale to as many destinations as you would like to have. Each script has
-a printout of each message that is sent, received, and counts the number
-of beacons that have currently been sent out. By default the number of
-bytes sent for each beacon is a random number between 0 and 1200. This
-can be changed at the line with “m = random.randint(0,1200)”. By default
-the beaconing interval is between 30 and 60 seconds. This can be changed
-at the line with “rsleep = random.randint(30,60)”. The client/server
-scripts have a comment on most of the lines with a brief description of
-its purpose if you are unsure of it.
+<h1 align="center">threat-tools</h1>
+
+This repository contains several scripts that will allow you to mimic a compromised system calling home to a command and control (C2) server. None of the tools exfiltrate any data and are designed to test your environment's ability to detect a wide range of C2 channels.
+
+## Table of Contents
+[Overview](#overview)</br>
+[Dependencies](#dependencies)</br>
+[`simple-listener.sh`](#simple-listenersh)</br>
+[`beacon-simulator.py`](#beacon-simulatorpy)</br>
+[`beacon-simulator.sh`](#beacon-simulatorsh)</br>
+[`simple-beacon.sh`](#simple-beaconsh)</br>
+[Python Script Pairs](#python-script-pairs)</br>
+[Running Tools in the Background with `screen`](#running-tools-in-the-background-with-screen)
+
+
+## Overview
+In order to run these scripts, you will need two systems: one to act as the compromised client sending the beacon, and one to act as the C2 server. The client will need to run one of the beacon simulator scripts, and the server will need to run one of the scripts to set up a listener. With exception to the scripts within `python-scripts`, all beacon simulators are meant to be used in tandem with `simple-listener.sh` running on the server. The only exception to this is if the client is running `simple-beacon.sh`, which can be used on its own without additional scripts running on the server.
+
+For the majority of users, `beacon-simulator.py` and `simple-listener.sh` will be sufficient to simulate a C2 channel. Alternative beacon simulators are available in the `shell-scripts` and `python-scripts` directories.
+
+## Dependencies
+In order to use the tools, you must have `ncat`, `hping3`, and `python3` installed on each of your systems. You will also need to install and use `pip3` to install `pycryptodome` if your system doesn't have it already.
```

Comment:
Let's split this up and include the dependencies under each script. I don't want to scare users away with install pycryptodome if they don't want to use the python scripts

Reply:
Done! Much easier this way. I included a section at the top of the README with an example setup as well including a step-by-step guide for new users on how to set up a channel with `simple-listener.sh` and `beacon-simulator.py`.